src/webpack/webpack.ts

```diff
@@ -1,104 +1,48 @@
-     */
-    libraryName?: string;
-    /**
-     * Allows a custom output file name to be used instead of the default [name]-bundle.js
-     */
-    outputFilename?: string;
-}
-
-/**
- * Shared function to create a webpack config for an entry point
- */
-export function createConfig(outPath: string, entryPoint: string, options: CustomWebpackOptions, ...plugins: webpack.Plugin[]) {
-    const config: webpack.Configuration = {
-        entry: entryPoint,
-        optimization: {minimize: !options || options.minify !== false},
-        output: {path: outPath, filename: `${options && options.outputFilename || '[name]-bundle'}.js`},
-        resolve: {extensions: ['.ts', '.tsx', '.js']},
-        module: {
-            rules: [
-                {test: /\.css$/, loader: 'style-loader'},
-                {test: /\.css$/, loader: 'css-loader'},
-                {test: /\.module.css$/, loader: 'css-loader', query: {modules: true, localIdentName: '[name]__[local]___[hash:base64:5]'}},
-                {test: /\.(png|jpg|gif|otf|svg)$/, use: [{loader: 'url-loader', options: {limit: 8192}}]},
-                {test: /\.tsx?$/, loader: 'ts-loader'}
-            ]
-        },
-        plugins: []
-    };
-
-    if (options && options.isLibrary === true) {
-        if (!!options.libraryName) {
-            config.output!.library = options.libraryName;
-        } else {
-            config.output!.library = '[name]';
-        }
-        config.output!.libraryTarget = 'umd';
-    }
-    if (plugins && plugins.length) {
-        config.plugins!.push.apply(config.plugins, plugins);
-    }
-
-    return config;
-}
+import * as webpackDevMiddleware from 'webpack-dev-middleware';
+import {WebpackMode} from '../types';
+import {getRootDirectory} from '../utils/getRootDirectory';
 
 /**
- * Provider temporarily requires an extra plugin to override index.html within provider app.json
- * Will be removed once the RVM supports relative paths within app.json files
+ * Executes Webpack.  Doubles as express-compatible middleware function to serve webpack modules.
+ *
+ * Wrapper will immediately terminate the server if the initial build fails.
+ *
+ * This is a wrapper around the webpack-dev-middleware utility.
  */
-export const manifestPlugin = (() => {
-    const {SERVICE_NAME, PORT} = getProjectConfig();
-
-    return new CopyWebpackPlugin([{
-        from: './res/provider/app.json',
-        to: '.',
-        transform: (content) => {
-            const config = JSON.parse(content);
-
-            if (typeof process.env.SERVICE_VERSION !== 'undefined' && process.env.SERVICE_VERSION !== '') {
-                config.startup_app.url = `https://cdn.openfin.co/services/openfin/${SERVICE_NAME}/` + process.env.SERVICE_VERSION + '/provider.html';
-                config.startup_app.autoShow = false;
+export async function executeWebpack(mode: WebpackMode, writeToDisk: boolean): Promise<RequestHandler> {
```

Comment:
This is now the only thing in here, rename file to `executeWebpack.ts`?

Reply:
Good catch, fixed